tests/MongoDB.EntityFrameworkCore.FunctionalTests/Mapping/ClrTypeMappingTests.cs

```diff
@@ -180,382 +180,400 @@ public ListSubclass(IEnumerable<T> items) : base(items)
+        {
+            var actual = db.Entities.FirstOrDefault();
 
-        Assert.NotNull(actual);
-        Assert.Equal(0, actual.anInt16);
+            Assert.NotNull(actual);
+            Assert.Equal(updated, actual.aByteArray);
+        }
     }
 
-    [Fact]
-    public void Int16_negative_read()
-    {
+    [Theory]
+    [InlineData(short.MaxValue, short.MinValue)]
+    [InlineData((short)-1, short.MaxValue)]
+    [InlineData((short)0, (short)-1)]
+    [InlineData((short)1, (short)0)]
+    [InlineData(short.MinValue, 1)]
+    public void Int16_read_update(short inserted, short updated)
+    {
+        database.DropCollection();
         var collection = database.CreateCollection<Int16Entity>();
+        var id = ObjectId.GenerateNewId();
+        collection.InsertOne(new Int16Entity {_id = id, anInt16 = inserted});
 
-        var expected = (Int16)_random.Next(Int16.MinValue, -1);
-        collection.InsertOne(new Int16Entity {_id = ObjectId.GenerateNewId(), anInt16 = expected});
-
-        using var db = SingleEntityDbContext.Create(collection);
-        var actual = db.Entities.FirstOrDefault();
-
-        Assert.NotNull(actual);
-        Assert.Equal(expected, actual.anInt16);
-    }
-
-    [Fact]
-    public void Int32_positive_read()
-    {
-        var collection = database.CreateCollection<Int32Entity>();
-
-        var expected = _random.Next(1, Int32.MaxValue);
-        collection.InsertOne(new Int32Entity {_id = ObjectId.GenerateNewId(), anInt32 = expected});
-
-        using var db = SingleEntityDbContext.Create(collection);
-        var actual = db.Entities.FirstOrDefault();
-
-        Assert.NotNull(actual);
-        Assert.Equal(expected, actual.anInt32);
-    }
-
-    [Fact]
-    public void Int32_zero_read()
-    {
-        var collection = database.CreateCollection<Int32Entity>();
+        using (var db = SingleEntityDbContext.Create(collection))
+        {
+            var actual = db.Entities.First(e => e._id == id);
 
-        collection.InsertOne(new Int32Entity {_id = ObjectId.GenerateNewId(), anInt32 = 0});
+            Assert.Equal(inserted, actual.anInt16);
+            actual.anInt16 = updated;
+            db.SaveChanges();
+        }
 
-        using var db = SingleEntityDbContext.Create(collection);
-        var actual = db.Entities.FirstOrDefault();
+        using (var db = SingleEntityDbContext.Create(collection))
+        {
+            var actual = db.Entities.First(e => e._id == id);
 
-        Assert.NotNull(actual);
-        Assert.Equal(0, actual.anInt32);
+            Assert.NotNull(actual);
+            Assert.Equal(updated, actual.anInt16);
+        }
     }
 
-    [Fact]
-    public void Int32_negative_read()
-    {
+    [Theory]
+    [InlineData(int.MaxValue, int.MinValue)]
+    [InlineData(-1, int.MaxValue)]
+    [InlineData(0, -1)]
+    [InlineData(1, 0)]
+    [InlineData(int.MinValue, 1)]
+    public void Int32_read_update(int inserted, int updated)
+    {
+        database.DropCollection();
```

Comment:
Our current pattern was that each test fixture creates it's own database and each test it's own collection if needed.
At the start of a run it clears all the previous DBs.

Rather than relying on the dropping you can just have it create a new collection passing in the args to CreateCollection.  I'd rather go this route as then you can see all the collections and databases from the last run and don't have to jump in/comment out/break-stop to get any kind of transient collection.
e.g.

```
var collection = database.CreateCollection<Int32Entity>(values: [inserted, updated]);
```

Which will create a uniquely named one that uses those args.